plenum/bls/bls_bft_replica_plenum.py

```diff
@@ -19,10 +21,14 @@ def __init__(self,
                  node_id,
                  bls_bft: BlsBft,
                  is_master,
+                 database_manager: DatabaseManager,
                  metrics: MetricsCollector = NullMetricsCollector()):
         super().__init__(bls_bft, is_master)
+        self._all_bls_latest_multi_sigs = None
         self.node_id = node_id
+        self._database_manager = database_manager
         self._signatures = {}
+        self._all_signatures = {}
```

Comment:
Do we really need `self._all_signatures` and `self._signatures`? (`self._all_bls_latest_multi_sigs` and `self._bls_latest_multi_sig`)?
Can we make `self._signatures` (`self._bls_latest_multi_sig`) to be a special case in  `self._all_signatures` (`self._all_bls_latest_multi_sigs`)?

Reply:
I have thought about it. This way we will always need to fetch the correct ledger from `_all_signatures` according to `PrePrepare`. On the other hand we have some code duplication -- but it is pretty obvious and we will remove it with the next release and there will be no problems.